src/sentry/models/dashboard_permissions.py

```diff
@@ -16,17 +29,27 @@ class DashboardPermissions(Model):
     __relocation_scope__ = RelocationScope.Organization
 
     is_creator_only_editable = models.BooleanField(default=False)
+    is_editable_by_everyone = models.BooleanField(default=True)
+    teams_with_edit_access = models.ManyToManyField(
+        "sentry.Team", through=DashboardPermissionsTeam, blank=True
+    )
+
     dashboard = models.OneToOneField(
         "sentry.Dashboard", on_delete=models.CASCADE, related_name="permissions"
     )
 
-    def has_edit_permissions(self, userId):
-        if not self.is_creator_only_editable:
+    def has_edit_permissions(self, user_id):
+        if self.is_editable_by_everyone:
             return True
-        return userId == self.dashboard.created_by_id
+        if user_id == self.dashboard.created_by_id:
+            return True  # Dashboard creator will always have edit perms
+        for team in self.teams_with_edit_access.all():
+            if user_id in team.get_member_user_ids():
```

Comment:
You have N queries here. If team assignments to dashboards get long this could get slow.

Reply:
Changed to this:
``` typescript
        return self.teams_with_edit_access.filter(
            organizationmemberteam__organizationmember__user_id=user_id
        ).exists()
```

Reply:
🎉 Very nice, much better.